### PR TITLE
Adding Event Prober lib that wrapps eventhub into an instance.

### PR DESCRIPTION
**Pull request by n3wscott**

### CODE REVIEW

File: pkg/eventshub/prober.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+	}
+	return nil
+}
+
+// SenderEventsFromURI configures a sender to send a url/yaml based events.
+func (p *EventProber) SenderEventsFromURI(uri string) {
+	p.senderOptions = append(p.senderOptions, InputYAML(uri))
+}
+
+// SenderEventsFromSVC configures a sender to send a url/yaml based events.
```
Comment (vaikas) [inline pkg/eventshub/prober.go:208]: is svcName with / without namespace. Maybe an example here?

File: pkg/eventshub/prober.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+		u, err := svc.Address(ctx, svcName)
+		if err != nil {
+			t.Error(err)
+		}
+		u.Path = path
+		p.senderOptions = append(p.senderOptions, InputYAML(u.String()))
+	}
+}
+
+// SenderFullEvents loads `count` cloudevents.FullEvent events with new IDs into a
```
Comment (vaikas) [inline pkg/eventshub/prober.go:220]: does it load or create new events? Seems like it generates them with different ids.

File: pkg/eventshub/prober.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+	for i := 0; i < count; i++ {
+		id := uuid.New().String()
+		event := cetest.MinEvent()
+		event.SetID(id)
+		p.ids = append(p.ids, id)
+		p.senderOptions = append(p.senderOptions, InputEvent(event))
+	}
+}
+
+// AsRef returns the short-named component as a KReference.
```
Comment (vaikas) [inline pkg/eventshub/prober.go:244]: nit: asKRef?

File: resources/svc/service.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	return func(ctx context.Context, t feature.T) {
+		if _, err := manifest.InstallLocalYaml(ctx, cfg); err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+// AsKRef returns a KReference for a Service without namespace.
```
Comment (vaikas) [inline resources/svc/service.go:51]: AsRef or AsKRef?

File: resources/svc/service_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2020 The Knative Authors
```
Comment (vaikas) [inline resources/svc/service_test.go:2]: nit copyright for the new files needs a bump.

File: test/example/prober_feature.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+	f := &feature.Feature{Name: "Prober with YAML"}
+
+	prober := eventshub.NewProber()
+
+	f.Setup("install recorder", prober.SenderInstall("recorder"))
+
+	_ = prober.SetTargetKRef(prober.AsKRef("recorder"))
+
+	// Locally, tell the test what to expect
+	if err := prober.ExpectYAMLEvents("https://raw.githubusercontent.com/cloudevents/conformance/v0.2.0/yaml/v1.0/v1_minimum.yaml"); err != nil {
```
Comment (vaikas) [inline test/example/prober_feature.go:60]: +1 to the url based fetching of the events.

**Review by vaikas — COMMENTED** [review]
there's also a few lint warnings.

File: pkg/eventshub/prober.go (hunk 2 above)
Comment (n3wscott) [reply pkg/eventshub/prober.go]: Reworded to say create.

File: pkg/eventshub/prober.go (hunk 1 above)
Comment (n3wscott) [reply pkg/eventshub/prober.go]: I added more to the comment, namespace comes from the env via the ctx in step

**Review by vaikas — APPROVED** [review]
/lgtm

**Pull request merged**